src/FocusTrap.ts

```diff
@@ -45,7 +45,7 @@ const FocusTrap: FocusTrapComponent = {
       default: true,
     },
     allowOutsideClick: {
-      type: Boolean,
+      type: [Boolean, Function],
```

Comment:
This probably needs an `as PropType<boolean | (e: MouseEvent) => boolean>`

Reply:
I made it `as PropType<FocusTrapComponentProps['allowOutsideClick']>` to prevent having to duplicate the function interface. Let me know if this doesn't seem right - I'm still learning my way around some of the nuances of TypeScript